llama_stack/providers/utils/common/data_schema_validator_mixin.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
Can datasetio abstract schema validation instead of creating mixins to be used by scoring and evals?

Reply:
@raghotham can you explain what you mean by this with an example? how do you imagine this validation happening specifically?